### ISSUE TITLE
Publish macosx pip wheels

### ISSUE THREAD

**Issue by jwnimmer-tri** [RobotLocomotion/drake#15958]
For our stable releases, we should upload a macOS-compatible wheel to PyPI.

- Highest priority is Python 3.9 (cp39-none, Homebrew default).
- Lower priority is Python 3.8 (cp38-none, Big Sur and Monterey default).
- We do not plan to ship Python 3.7 (cp37-none, Catalina default).

It's probable that we should improve our whl release automation, testing, and CI prior to attempting this.

**Comment by mwoehlke-kitware** [comment]
FYI, I'm not sure how to build 3.8 wheels on Big Sur. We would need to modify the build environment somehow so that 3.8 is the detected Python. Right now, CI is using 3.9. Also, we'll probably need separate CI jobs for each wheel.

#16872 is a first pass at implementing this. As of writing (including some not-yet-pushed-but-will-be-soon changes), the resulting wheel still depends on:
- /usr/local/opt/gcc/lib/gcc/11/libgfortran.5.dylib
- /usr/local/opt/gcc/lib/gcc/11/libquadmath.0.dylib
- /usr/local/opt/openblas/lib/libopenblas.0.dylib
- /usr/local/opt/nlopt/lib/libnlopt.0.dylib
- /usr/local/opt/glib/lib/libglib-2.0.0.dylib
- /usr/local/opt/gettext/lib/libintl.8.dylib

**Comment by jwnimmer-tri** [comment]
It's fine to defer 3.8.  We should focus on 3.9 and get that shipped in a stable release, documented on the website, fully debugged, etc.  After that, we can circle back to 3.8 (or return this to the backlog and switch to a new issue).

**Comment by mwoehlke-kitware** [comment]
Yup. Mostly I wanted to post the list of "stuff we currently need from homebrew" somewhere, and noticed that while I was here.

BTW, I just switched the macOS wheel builds to always building fmt/spdlog from source, using bazel, like manylinux already does. Tested locally; going to ask for CI jobs shortly.

**Comment by jwnimmer-tri** [comment]
Drake Jenkins is now uploading nightly builds of the macOS wheel every evening.

Here's how users can test the prototype nightlies:

```
$ wget https://drake-packages.csail.mit.edu/drake/nightly/drake-latest-cp39-cp39-macosx_11_0_x86_64.whl
$ python3 -m venv env
$ env/bin/pip install drake-latest-cp39-cp39-macosx_11_0_x86_64.whl
$ source env/bin/activate
(env) $ python3 -c 'import pydrake.all'
```

Our hope is to get some user feedback and testing in the next couple weeks, and then ship stable macOS wheels starting in our next release Drake v1.5.0 scheduled for mid-July.

**Comment by jwnimmer-tri** [comment]
I think I'll say we've reached the time limit waiting for any user feedback.

At this point, the only remaining work is ... (help me out here...)

(1) Document the nightly wheel candidates at https://drake.mit.edu/pip.html and https://drake.mit.edu/installation.html.

(2) Document the monthly release process changes to add macOS wheels.

Is that correct?  Anything else?

**Comment by mwoehlke-kitware** [comment]
> the only remaining work is [...]

Sounds right.

**Comment by jwnimmer-tri** [comment]
On (1), I remember now that #17413 has already started on the documentation.  Great.

What does the plan for (2) look like?  Perhaps a "Build with Parameters" Jenkins job (taking a git sha and version number) for a Mac Wheel build -- where the release manager requests a build as part of the release process, and can manually download the whl from a temporary location in S3?

**Comment by mwoehlke-kitware** [comment]
>  Perhaps a "Build with Parameters" Jenkins job (taking a git sha and version number) for a Mac Wheel build -- where the release manager requests a build as part of the release process, and can manually download the whl from a temporary location in S3?

On that front, see https://drake-jenkins.csail.mit.edu/view/Mac%20Big%20Sur%20Unprovisioned/job/mac-big-sur-unprovisioned-clang-wheel-staging-snopt-mosek-release/16/console (first total success of said job). The artifact (from that specific build) can be fetched from https://drake-packages.csail.mit.edu/drake/staging/drake-1.6a2-cp39-cp39-macosx_11_0_x86_64.whl. Obviously, that second URI will change with the version number.

This is still waiting on robotlocomotion/drake-ci#173 to land, but hopefully that will be "soon".

**Comment by jwnimmer-tri** [comment]
From f2f: the staging job should be available and ready now.  The nightly build whl filename output is almost done.
